src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs

```diff
@@ -204,12 +204,13 @@ protected virtual void WriteCtors()
 
         protected void WriteRestClientAssignments()
         {
-            // write assignment statements of the rest clients of this resource
-            foreach (var client in This.RestClients)
-            {
-                var subscriptionParamString = client.Parameters.Any(p => p.Name.Equals("subscriptionId")) ? ", Id.SubscriptionId" : string.Empty;
-                _writer.Line($"{GetRestClientVariableName(client)} = new {client.Type.Name}({ClientDiagnosticsField}, {PipelineProperty}, {ClientOptionsProperty}{subscriptionParamString}, {BaseUriField});");
-            }
+            Func<MgmtRestClient, string> getSubId = (restClient) => { return restClient.Parameters.Any(p => p.Name.Equals("subscriptionId")) ? ", Id.SubscriptionId" : string.Empty; };
+            WriteRestClientConstructionForResource(_resource, This.RestClients, getSubId, ClientDiagnosticsField, ClientOptionsProperty, PipelineProperty, BaseUriField, "new ", false);
+        }
+
+        private string ConvertToVariableName(string str)
+        {
+            return char.ToLower(str[0]) + str.Substring(1);
```

Comment:
Looks like this function is not used anywhere, could we remove it?
And we have similar one in StringExtensions `FirstLetterToUpperCase`

Reply:
`FirstLetterToLowerCase` ~

Reply:
removed as this wasn't needed anymore